EIPS/eip-4200.md

```diff
@@ -39,17 +39,21 @@ We introduce two new instructions on the same block number [EIP-3540](./eip-3540
 
 1. `RJUMP` (0x5c)
 2. `RJUMPI` (0x5d)
+3. `RJUMPV` (0x5e)
 
-If the code is legacy bytecode, both of these instructions result in an *exceptional halt*. (*Note: This means no change to behaviour.*)
+If the code is legacy bytecode, all of these instructions result in an *exceptional halt*. (*Note: This means no change to behaviour.*)
 
 If the code is valid EOF1:
 
-1. `RJUMP relative_offset` (0x5c), sets the `PC` to `PC_post_instruction + relative_offset`.
-2. `RJUMPI relative_offset` (0x5d), pops a value (`condition`) from the stack, and sets the `PC` to `PC_post_instruction + ((condition == 0) ? 0 : relative_offset)`.
+1. `RJUMP relative_offset` sets the `PC` to `PC_post_instruction + relative_offset`.
+2. `RJUMPI relative_offset` pops a value (`condition`) from the stack, and sets the `PC` to `PC_post_instruction + ((condition == 0) ? 0 : relative_offset)`.
+3. `RJUMPV count relative_offset+` pops a value (`case`) from the stack, and sets the `PC` to `PC_post_instruction + ((case >= count) ? 0 : relative_offset[case])`.
 
 The immediate argument `relative_offset` is encoded as a 16-bit **signed** (two's-complement) big-endian value. Under `PC_post_instruction` we mean the `PC` position after the entire immediate value.
 
-We also extend the validation algorithm of [EIP-3670](./eip-3670.md) to verify that each `RJUMP`/`RJUMPI` has a `relative_offset` pointing to an instruction. This means it cannot point to an immediate data of `PUSHn`/`RJUMP`/`RJUMPI`. It cannot point outside of code bounds. It is allowed to point to a `JUMPDEST`, but is not required to.
+The immediate encoding of `RJUMPV` is more special: the 8-bit `count` value incremented by 1 determines the number of `relative_offset` values following. Based on this it can be seen that the encoding of `RJUMPV` must have at least one `relative_offset` and thus it will take at minimum 4 bytes. Furthermore, the `case >= count` condition falling through means that in many use cases one would place the *default* path following the `RJUMPV` instruction.
```

Comment:
why incremented by 1?

Reply:
I guess to avoid empty jump table.

`RJUMPV 0` could be a no-op or rejected at validation.

Reply:
> I guess to avoid empty jump table.

Yes.

> RJUMPV 0 could be a no-op or rejected at validation.

Yes that's an option.

Reply:
Is there a reason to not just reject `count=0`?

Reply:
In the current description count is always `+1`d, so this removes the need for a special case of "reject count 0". We can go either way.

Reply:
We can also keep `count=0` as yet another NOP-style instruction if that's better.